### PR TITLE
Updated TACoApplication deployment

### PR DESCRIPTION
**Pull request by cygnusv**
**Type of PR:**
- [ ] Bugfix
- [ ] Feature
- [ ] Documentation
- [x] Other

**Required reviews:** 
- [ ] 1
- [x] 2
- [ ] 3

**What this does:**
> High-level idea of the changes introduced in this PR. 
> List relevant API changes (if any), as well as related PRs and issues.

**Issues fixed/closed:**
> - Fixes #...

**Why it's needed:**
> Explain how this PR fits in the greater context of the NuCypher Network.
> E.g., if this PR address a `nucypher/productdev` issue, let reviewers know!

**Notes for reviewers:**
> What should reviewers focus on? 
> Is there a particular commit/function/section of your PR that requires more attention from reviewers?

### CODE REVIEW

File: deployment/artifacts/mainnet.json [hunk 1]
```diff
@@ -149,7 +149,7 @@
             "deployer": "0xFfFd7092685bDeeBD121D1A0FEA3c349114Cce50"
         },
         "TACoApplication": {
-            "address": "0x347CC7ede7e5517bD47D20620B2CF1b406edcF07",
+            "address": "0x98F643B32b5A3513929Dfd3F2E3e39F7376659DB",
```
Comment (derekpierre) [inline deployment/artifacts/mainnet.json:152]: Was this deployed as a proxy upgrade? I thought TACoApplication was proxied, in which case the address shouldn't change.

Comment (derekpierre) [reply deployment/artifacts/mainnet.json:152]: ^ @cygnusv

Comment (cygnusv) [reply deployment/artifacts/mainnet.json:152]:
```suggestion
            "address": "0x347CC7ede7e5517bD47D20620B2CF1b406edcF07",
```

Comment (derekpierre) [reply deployment/artifacts/mainnet.json]: I'm wondering if there may be an issue in the deployment script itself if this needed to be manually reverted. I believe for proxy deployment upgrades the `Deployer` typically handles not updating the address but updating the ABI in the resulting contract registry.

I'll try to take a deeper look at the deployment script.

(cc @KPrasch )

**Review by KPrasch — APPROVED** [review]

**Review by derekpierre — CHANGES_REQUESTED** [review]
Just putting this marker for now until we take a deeper look at the deployment script.

File: deployment/artifacts/mainnet.json (hunk 1 above)
Comment (derekpierre) [reply deployment/artifacts/mainnet.json]: I think I get it now.

On mainnet we can't actually upgrade the proxied contracts ourselves; that can only be done by the Council Multisig. As a result we only deploy implementation contracts, and then have the Council subsequently upgrade the proxy to point to the implementation contract. However, our deployment scripts are not accustomed to the proxy upgrade process being split into two separate operations.

As a result, when we deploy the implementation contract on the `mainnet,` the contract registry address gets changed even for a proxy contract. 

For now we have to be careful not to update the addresses in the contract registry, and maybe in the future we can come up with an option to not have the registry update the address of the contract.

In the context of this PR we are good now that the address change has been reverted to the original address.

That being said, we should not merge this PR into `main` until the Council has upgraded the TACoApplication proxy contract (unless that has already happened).

**Review by derekpierre — APPROVED** [review]
🎸

**Comment by derekpierre** [comment]
Council already upgraded the proxy contract - https://etherscan.io/tx/0xc3adf42427bbf4f25026da756a02e47317e11bd6ae6cebe8303b38b72f660370

**Pull request merged**